pkg/apis/pipeline/v1beta1/resource_types.go

```diff
@@ -70,6 +70,16 @@ type TaskResources struct {
 	// DeclaredPipelineResources to the input PipelineResources required by the Task.
 	// +listType=atomic
 	Outputs []TaskResource `json:"outputs,omitempty"`
+	// Limits describes the maximum amount of compute resources allowed.
+	// More info: https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/
+	// +optional
+	Limits v1.ResourceList `json:"limits,omitempty"`
```

Comment:
Also, as detailed in the TEP, these fields should be added to TaskRun and PipelineRun.TaskRunSpecs too.

Reply:
yap. planned to deliver as a follow-up PR (focusing on "runtime overwriting" part)

Reply:
closed - by updating comments + docs